ext/civiimport/Civi/Api4/Event/Subscriber/ImportSubscriber.php

```diff
@@ -112,4 +112,70 @@ public function onApiAuthorize(AuthorizeEvent $event): void {
     }
   }
 
+  /**
+   * Get an array of FormBuilder forms for viewing imports.
+   *
+   * @param \Civi\Core\Event\GenericHookEvent $event
+   *
+   * @throws \CRM_Core_Exception
+   *
+   * @noinspection PhpUnused
+   */
+  public static function on_civi_afform_get(GenericHookEvent $event): void {
+    // Early return if forms are not requested -  copied from eck code - correct?
+    if ($event->getTypes && !in_array('form', $event->getTypes, TRUE)) {
+      return;
+    }
+
+    $importForms = self::getImportForms();
+    if (!empty($importForms) && $importForms !== $event->afforms) {
+      $event->afforms = array_merge($event->afforms ?? [], $importForms);
+    }
+  }
+
+  /**
+   * Get an array of FormBuilder forms for viewing imports.
+   *
+   * @return array
+   *
+   * @throws \CRM_Core_Exception
+   */
+  public static function getImportForms(): array {
+    $cacheKey = 'civiimport_forms_' . \CRM_Core_Config::domainID() . '_' . (int) \CRM_Core_Session::getLoggedInContactID();
+    if (\Civi::cache('metadata')->has($cacheKey)) {
+      return \Civi::cache('metadata')->get($cacheKey);
+    }
+    $forms = [];
+    try {
+      $importSearches = SearchDisplay::get()
+        ->addWhere('saved_search_id.name', 'LIKE', 'Import\_Summary\_%')
+        ->addWhere('saved_search_id.expires_date', '>', 'now')
+        ->addSelect('name', 'label')
+        ->execute();
+      foreach ($importSearches as $importSearch) {
+        $userJobID = str_replace('Import_Summary_', '', $importSearch['name']);
+        $forms[$importSearch['name']] = [
+          'name' => $importSearch['name'],
+          'type' => 'search',
+          'title' => $importSearch['label'],
+          'base_module' => E::LONG_NAME,
+          'is_dashlet' => FALSE,
+          'is_public' => FALSE,
+          'is_token' => FALSE,
+          'permission' => 'access CiviCRM',
+          'server_route' => 'civicrm/imports/' . $importSearch['name'],
```

Comment:
I think these ought to be removed since the form is meant to be embedded not on its own page.

Reply:
Done